### PR TITLE
chore: improvements to build multiple distribution flavors

### PR DESCRIPTION
**Pull request by phiz71**
Use profiles to manage the different dependencies in release distrib, dev distrib, EE distrib, ...

Improvement:
* Instead of using maven-dependency-plugin, rely on pom dependencies and some configuration in assembly files to build the final binaries. It allows to use `mvn versions:update-properties`, to keep distribution up to date automatically
* No need to create a tar.gz, since it will be unpack in dockerFile. Just use a folder.

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-pdfbxwsumf.chromatic.com)

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/multiple-distribution/index.html)
_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-standalone/gravitee-apim-rest-api-standalone-distribution/gravitee-apim-rest-api-standalone-distribution-zip/src/main/assembly/plugin-assembly-standalone.xml [hunk 1]
```diff
@@ -30,112 +30,21 @@
     <includeBaseDirectory>true</includeBaseDirectory>
 
     <fileSets>
-        <!-- Copy startup scripts and config files -->
         <fileSet>
-            <directory>../target/distribution/config</directory>
-            <outputDirectory>config</outputDirectory>
+            <directory>../target/gravitee-apim-rest-api-distribution-${project.version}</directory>
```
Comment (gaetanmaisse) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-standalone/gravitee-apim-rest-api-standalone-distribution/gravitee-apim-rest-api-standalone-distribution-zip/src/main/assembly/plugin-assembly-standalone.xml]: It's a bit painful to have the version in the folder name as it forces us to have 1 IntelliJ config per version. Can we remove it?

File: gravitee-apim-distribution/pom.xml [hunk 2]
```diff
@@ -38,16 +38,15 @@
         To add a new plugin in the bundle:
          1. add the version in the <properties> section.
             please respect alphabetic order and the separation between Gateway only, Management only and both
-         2. update the maven-dependency-plugin so the new plugin can be copied in the right temp folder (used in the packaging).
-            It can be apim-plugins, connectors, fetchers, policies, ... or a new kind of plugins.
-         3. update the Gateway and/or the management API assembly.xml files if necessary
+         2. add the plugin in the <dependencies> section with <type>zip</type> and <scope>runtime</scope>.
```
Comment (gaetanmaisse) [inline gravitee-apim-distribution/pom.xml:41]: 👌🏻 it looks clear and understandable even for non Maven expert

File: gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-distribution/gravitee-apim-gateway-standalone-distribution-zip/src/main/assembly/plugin-assembly-standalone.xml [hunk 3]
```diff
@@ -29,64 +30,19 @@
     <includeBaseDirectory>true</includeBaseDirectory>
 
     <fileSets>
-        <!-- Copy startup scripts and config files -->
         <fileSet>
-            <directory>../target/distribution/config</directory>
-            <outputDirectory>config</outputDirectory>
+            <directory>../target/gravitee-apim-gateway-distribution-${project.version}</directory>
```
Comment (gaetanmaisse) [inline gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-distribution/gravitee-apim-gateway-standalone-distribution-zip/src/main/assembly/plugin-assembly-standalone.xml]: Same as for the rest api:
> It's a bit painful to have the version in the folder name as it forces us to have 1 IntelliJ config per version. Can we remove it?

File: gravitee-apim-rest-api/docker/Dockerfile-dev [hunk 4]
```diff
@@ -60,29 +58,3 @@ HEALTHCHECK --interval=5s --timeout=3s --retries=6 --start-period=30s \
...
-	&& chgrp -R 0 ${GRAVITEEIO_HOME} \
-    && chmod -R g=u ${GRAVITEEIO_HOME}
-
-WORKDIR ${GRAVITEEIO_HOME}
-EXPOSE 8083
-ENTRYPOINT ["./bin/gravitee"]
-VOLUME ${GRAVITEEIO_HOME}/logs
-CMD ["./bin/gravitee"]
-HEALTHCHECK --interval=5s --timeout=3s --retries=6 --start-period=30s \
-            CMD curl --user admin:adminadmin --fail http://localhost:18083/_node/health || exit 1
```
Comment (gaetanmaisse) [inline gravitee-apim-rest-api/docker/Dockerfile-dev]: It's a legacy thing that should have been removed earlier?

Comment (phiz71) [reply gravitee-apim-rest-api/docker/Dockerfile-dev]: Yes, it should be part of another PR on master. It comes from the merge of the "performance branch"

File: gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-distribution/gravitee-apim-gateway-standalone-distribution-zip/src/main/assembly/plugin-assembly-standalone.xml (hunk 3 above)
Comment (phiz71) [reply gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-distribution/gravitee-apim-gateway-standalone-distribution-zip/src/main/assembly/plugin-assembly-standalone.xml]: you're right, I will change it

File: gravitee-apim-distribution/pom.xml (hunk 2 above)
Comment (phiz71) [reply gravitee-apim-distribution/pom.xml:41]: 🙏🏻

File: gravitee-apim-rest-api/gravitee-apim-rest-api-standalone/gravitee-apim-rest-api-standalone-distribution/gravitee-apim-rest-api-standalone-distribution-zip/src/main/assembly/plugin-assembly-standalone.xml (hunk 1 above)
Comment (phiz71) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-standalone/gravitee-apim-rest-api-standalone-distribution/gravitee-apim-rest-api-standalone-distribution-zip/src/main/assembly/plugin-assembly-standalone.xml]: you're right, I will change it

File: gravitee-apim-rest-api/docker/Dockerfile-dev (hunk 4 above)
Comment (phiz71) [reply gravitee-apim-rest-api/docker/Dockerfile-dev]: https://github.com/gravitee-io/gravitee-api-management/pull/1073

**Review by gaetanmaisse — APPROVED** [review]

**Pull request merged**